cli/cmd/deploy/deploy.go

```diff
@@ -356,6 +336,106 @@ func verifyAccess(ctx context.Context, c *client.Client, githubURL string) (*adm
 	return ghRes, nil
 }
 
+func inferSourceConnectors(projectPath string) (map[connector][]string, error) {
```

Comment:
Let's pull this out into `runtime/compilers/rillv1beta`, maybe call it `ExtractConnectors` and return just `([]*Connector, error)` (no source names). Keeps it simpler – and makes it easy to extend for connectors defined in `rill.yaml` later.